### PR TITLE
Fix: GraphQL validates repeatable component attributes correctly

### LINKED ISSUE #17995: GraphQl returns "Invalid parameter X" on any filter
### Required System information

Node.js version: v18.17.1
YARN version: 1.22.19
Strapi version: 4.13.3
Database: SQLITE
Operating system: Windows 10
Is your project Javascript or Typescript: JavaScript

### Describe the bug

Unable to resolve graphql queries having "filter" property 

### Steps to reproduce the behavior

1. Trying to get data form table products with this query:
```
#import "/apollo/fragments/shared/Technical.graphql"

query products(
  $locale: I18NLocaleCode
  $limit: Int!
  $start: Int!
  $brands: [String!]
  $categories: [String!]
  $slug: String
  $id: ID
  $ids: [ID]
  $notSlug: String
  $notId: ID
  $name: String
  $priceMinimal: Float
  $priceMaximal: Float
  $bestseller: Boolean
  $sale: Boolean
  $novelty: Boolean
) {
  products(
    locale: $locale
    pagination: { start: $start, limit: $limit }
    sort: "createdAt:desc"
    filters: {
      id: { eq: $id, in: $ids, ne: $notId }
      slug: { eq: $slug, ne: $notSlug }
      name: { containsi: $name }
      brand: { slug: { in: $brands } }
      categories: { slug: { in: $categories } }
      price: { between: [$priceMinimal, $priceMaximal] }
      active: { eq: true }
      technical: {
        novelty: { eq: $novelty }
        sale: { eq: $sale }
        bestseller: { eq: $bestseller }
      }
    }
  ) {
    data {
      id
      attributes {
        name
        slug
        status
        price
        priceSale
        gallery {
          data {
            attributes {
              formats
              url
              caption
              alternativeText
            }
          }
        }
        picture {
          data {
            attributes {
              formats
              url
              caption
              alternativeText
            }
          }
        }
        brand {
          data {
            attributes {
              name
              slug
            }
          }
        }
        categories {
          data {
            id
            attributes {
              name
              slug
            }
          }
        }
        technical {
          ...Technical
        }
      }
    }
    meta {
      pagination {
        total
        page
        pageCount
        pageSize
      }
    }
  }
}
```

2. Sending params in my request:

```
{
    "limit": 12,
    "start": 0,
    "locale": "en",
    "priceMinimal": 0,
    "priceMaximal": 99999
} 
```

3. Getting an error:

```
"message": "Invalid parameter id",
```

### Expected behavior

Receiving an answer from the server which contains data[] with product information.

### Additional context

Then I remove filter { } from the query - everything works fine.

schema.graphql contains the needed fields of course:

```
input ProductFiltersInput {
    active: BooleanFilterInput
    and: [ProductFiltersInput]
    announcement: StringFilterInput
    brand: BrandFiltersInput
    categories: CategoryFiltersInput
    createdAt: DateTimeFilterInput
    description: StringFilterInput
    features: ComponentSharedFeatureFiltersInput
    id: IDFilterInput
    locale: StringFilterInput
    localizations: ProductFiltersInput
    name: StringFilterInput
    not: ProductFiltersInput
    or: [ProductFiltersInput]
    price: FloatFilterInput
    priceSale: FloatFilterInput
    publishedAt: DateTimeFilterInput
    seo: ComponentSharedSeoFiltersInput
    sitemap_exclude: BooleanFilterInput
    slug: StringFilterInput
    status: StringFilterInput
    technical: ComponentSharedTechnicalFiltersInput
    updatedAt: DateTimeFilterInput
}
```

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Fixes a typo in the graphql component resolver

### Why is it needed?

Query arguments for graphql components were targeting the parent model for validation rather than the component model

### How to test it?

- create a content type with a repeatable component
- the component should have an attribute with a name not present in the content type
- run a query with a filter on that attribute of the component
- before this PR: an error would have been thrown about an invalid parameter
- with this PR: the attribute should be accepted
 
Also added a simple test that fails without the PR and passes with the PR.

### Related issue(s)/PR(s)

Fixes #17995 
Related fix #18180

### CODE REVIEW

**Review by christiancp100 — DISMISSED** [review]

**Comment by derrickmehaffy** [comment]
Building experimental: https://github.com/strapi/strapi/actions/runs/6734980357

**Review by alexandrebodin — DISMISSED** [review]
LGTM. No problems on relations & dz ?

File: api-tests/plugins/graphql/components.test.api.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    },
+  },
+  singularName: 'label',
+  pluralName: 'labels',
+  displayName: 'Label',
+  description: '',
+  collectionName: '',
+};
+
+describe('Test Graphql Relations API End to End', () => {
```
Comment (innerdvations) [inline api-tests/plugins/graphql/components.test.api.js:54]:
```suggestion
describe('Test Graphql Components API End to End', () => {
```

**Review by alexandrebodin — APPROVED** [review]

**Comment by alexandrebodin** [comment]
Haven't done any QA*

**Comment by innerdvations** [comment]
> LGTM. No problems on relations & dz ?

Nope, they only query on specific ids so no issues there.

**Review by Bassel17 — APPROVED** [review]
LGTM! QAed as well

**Pull request merged**